sklearn/inspection/_partial_dependence.py

```diff
@@ -477,6 +497,10 @@ def partial_dependence(
                 "'decision_function'. Got {}.".format(response_method)
             )
 
+    if sample_weight is not None:
+        sample_weight = column_or_1d(sample_weight)
+        check_consistent_length(X, sample_weight)
```

Comment:
We should have a `_check_sample_weight` in the `sklearn.utils.validation` that is doing this check

Reply:
Should I set `only_non_negative=True` here (param of `_check_sample_weight`)?